### PR TITLE
Microchip MEC172x Add clock control driver

### PR DESCRIPTION
**Pull request by scottwcpg**
This PR depends upon PR 36864.
This PR adds a clock control driver for MEC172x.  Code configuring the 32 KHz clock sources moved from SoC layer to this driver.  A device tree node for MEC172x PCR was added with 32 KHz clock source, CPU clock divider, and PCR slow clock divider properties.

### CODE REVIEW

File: boards/arm/mec172xevb_assy6906/mec172xevb_assy6906_defconfig [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#
+# Copyright (c) 2019, Intel Corporation
```
Comment (sjg20) [inline boards/arm/mec172xevb_assy6906/mec172xevb_assy6906_defconfig:2]: Intel?

File: soc/arm/microchip_mec/mec172x/Kconfig.defconfig.series [hunk 2]
```diff
@@ -19,4 +19,10 @@ source "soc/arm/microchip_mec/mec172x/Kconfig.defconfig.mec172x*"
 config CORTEX_M_SYSTICK
 	default y
 
+config CLOCK_CONTROL_MCHP_XEC
+	default y
+	depends on CLOCK_CONTROL
+	help
+	  Enable support for Microchip XEC clock controller driver.
```
Comment (sjg20) [inline soc/arm/microchip_mec/mec172x/Kconfig.defconfig.series:26]: clock-controller driver

Can you add a bit of help about what features the driver has? Kconfig help is supposed to be 4+ lines as I understand it

**Review by sjg20 — CHANGES_REQUESTED** [review]
Looks reasonable.

I'm not a fan of the #ifdefs in the the header to select different values, but I suppose splitting these bits out into separate files for each SoC would be even more confusing.

File: boards/arm/mec172xevb_assy6906/mec172xevb_assy6906_defconfig (hunk 1 above)
Comment (scottwcpg) [reply boards/arm/mec172xevb_assy6906/mec172xevb_assy6906_defconfig]: done

File: soc/arm/microchip_mec/mec172x/Kconfig.defconfig.series (hunk 2 above)
Comment (scottwcpg) [reply soc/arm/microchip_mec/mec172x/Kconfig.defconfig.series]: Done

**Comment by scottwcpg** [comment]
Closing due to agreement reached on MEC172x SoC header files will be located in zephyr soc. All drivers will use these headers.

**Pull request closed without merging**